packages/@aws-cdk/aws-rds/lib/parameter-group.ts

```diff
@@ -93,56 +84,85 @@ export interface ParameterGroupProps {
-    });
+      public addParameter(_key: string, _value: string): boolean {
+        return false;
+      }
+    }
 
-    this.parameterGroupName = resource.ref;
+    return new Import(scope, id);
   }
-}
 
-/**
- * Construction properties for a ClusterParameterGroup
- */
-// tslint:disable-next-line:no-empty-interface
-export interface ClusterParameterGroupProps extends ParameterGroupProps {
+  private readonly parameters: { [key: string]: string };
+  private readonly family: string;
+  private readonly description?: string;
 
-}
-/**
- * A cluster parameter group
- *
- * @resource AWS::RDS::DBClusterParameterGroup
- */
-export class ClusterParameterGroup extends ParameterGroupBase {
-  /**
-   * The name of the parameter group
-   */
-  public readonly parameterGroupName: string;
+  private clusterCfnGroup?: CfnDBClusterParameterGroup;
+  private instanceCfnGroup?: CfnDBParameterGroup;
 
-  constructor(scope: Construct, id: string, props: ClusterParameterGroupProps) {
-    super(scope, id, props.parameters);
+  constructor(scope: Construct, id: string, props: ParameterGroupProps) {
+    super(scope, id);
 
-    const resource = new CfnDBClusterParameterGroup(this, 'Resource', {
-      description: props.description || `Cluster parameter group for ${props.family}`,
-      family: props.family,
-      parameters: Lazy.anyValue({ produce: () => this.parameters }),
-    });
+    this.family = props.family;
+    this.description = props.description;
+    this.parameters = props.parameters ?? {};
+  }
 
-    this.parameterGroupName = resource.ref;
+  public bindToCluster(_options: ParameterGroupClusterBindOptions): ParameterGroupClusterConfig {
+    if (!this.clusterCfnGroup) {
+      const parentScope = this.instanceCfnGroup ?? this;
+      this.clusterCfnGroup = new CfnDBClusterParameterGroup(parentScope, 'Resource', {
+        description: this.description || `Cluster parameter group for ${this.family}`,
+        family: this.family,
+        parameters: Lazy.anyValue({ produce: () => this.parameters }),
+      });
+    }
+    return {
+      parameterGroupName: this.clusterCfnGroup.ref,
+    };
+  }
+
+  public bindToInstance(_options: ParameterGroupInstanceBindOptions): ParameterGroupInstanceConfig {
+    if (!this.instanceCfnGroup) {
+      const parentScope = this.clusterCfnGroup ?? this;
+      this.instanceCfnGroup = new CfnDBParameterGroup(parentScope, 'Resource', {
+        description: this.description || `Parameter group for ${this.family}`,
+        family: this.family,
+        parameters: Lazy.anyValue({ produce: () => this.parameters }),
+      });
+    }
+    return {
+      parameterGroupName: this.instanceCfnGroup.ref,
+    };
+  }
+
+  /**
+   * Add a parameter to this parameter group
+   *
+   * @param key The key of the parameter to be added
+   * @param value The value of the parameter to be added
+   */
+  public addParameter(key: string, value: string): boolean {
```

Comment:
what's the use case where the return type would be used?
at a cursory glance, most of the `addXyZ` methods return void or a non-primitive type

Reply:
Sure, but if you're going to return `void`, you might as well return `boolean` (I don't see any downsides in making it return `boolean` compared to `void`, seems like pure upside to me).